stdlib/public/runtime/DynamicCast.cpp

```diff
@@ -712,6 +712,14 @@ struct ObjCBridgeMemo {
 };
 #endif
 
+static bool avoidOptionalsInAnyHashable() {
+  // Unexpected nulls are fatal in Swift 5.4 and not before
+  // AnyHashable avoids optionals before Swift 5.4
+  // TODO: This is admittedly awkward, but I need to get this changed quickly
+  // I'll come back and clean this up later.
+  return !unexpectedNullIsFatal();
```

Comment:
I think this should just return true for now, without doing a linked-on-or-after check -- otherwise currently working code that bridges optional-keyed dictionaries through AnyHashable will subtly break when recompiled with the SDKs that ship with 5.4.

The issue with Optionals vs AnyHashable is tracked in https://bugs.swift.org/browse/SR-9047 (with a WIP fix in https://github.com/apple/swift/pull/20523). The fix used to be blocked by our inability to add versioned conformances. Slava resolved that in https://github.com/apple/swift/pull/34565, so I'm hoping we can land the fix in the release after 5.4.

Once SR-9047 gets fixed, we can add the link-on-or-after check to get the better casting behavior without breaking dictionary bridging.

Reply:
I agree.  Done.